src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1148,20 +1148,20 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
                     var temp = _factory.StoreToTemp(
                         argument,
                         out BoundAssignmentOperator assignment,
-                        refKind: paramRefKind == RefKind.In ? RefKind.In : argRefKind);
+                        refKind: paramRefKind is RefKind.In or RefKind.RefReadOnlyParameter ? RefKind.In : argRefKind);
```

Comment:
>refKind: paramRefKind is RefKind.In or RefKind.RefReadOnlyParameter ? RefKind.In : argRefKind); [](http://example.com/codeflow?start=24&length=95)

This looks inconsistent with how ```GetEffectiveArgumentRefKinds``` adjusts ref kinds.

Reply:
> I assume you mean it doesn't pass ```StrictIn``` in case ```argRefKind == None```?

No, I mean it doesn't pass ```StrictIn``` in case ```argRefKind != None``` 

> It didn't do that previously for in either

I think that was just an omission.

> and I don't see why it needs to be doing that.

Usage of ```StrictIn``` is important to avoid unexpected references to copies. If we look at how ```SpillSequenceSpiller``` spills arguments to locals (which, is effectively what this code is doing), we see that the adjusted ref kind is passed to ```StoreToTemp``` there. I can walk you through the code offline.